src/dmd/dtoh.d

```diff
@@ -1278,7 +1278,24 @@ public:
             printf("[AST.TypeIdentifier enter] %s\n", t.toChars());
             scope(exit) printf("[AST.TypeIdentifier exit] %s\n", t.toChars());
         }
+        if (t.idents.length)
+            buf.writestring("typename ");
+
         buf.writestring(t.ident.toChars());
+
+        foreach (arg; t.idents)
+        {
+            buf.writestring("::");
+
+            import dmd.root.rootobject;
+            // Is this even possible?
```

Comment:
I doubt it.

Reply:
Guess that's a consequence of inheriting from `TypeQualified` which is also used for template instance parameters.

I think i'll rather keep it just to be sure.